src/tribler/core/logger/tests/test_logger.py

```diff
@@ -18,23 +17,13 @@ def test_get_logger_config_path():
         assert config_path == Path('/x/y/z/tribler_source/tribler/core/logger/logger.yaml')
 
 
-@patch('tribler.core.logger.logger.logger')
-@patch('sys.stdout')
-@patch('sys.stderr')
-@patch('builtins.print')
-@patch('logging.basicConfig')
-def test_setup_logging_no_config(basic_config: Mock, print_: Mock, stderr: Mock, stdout: Mock, logger: Mock):
-    config_path = MagicMock()
-    config_path.exists.return_value = False
-    config_path.__str__.return_value = '<config-path>'
-
-    setup_logging('<app-mode>', '<log-dir>', config_path)
+@patch('yaml.safe_load')
+def test_setup_logging_no_config(mocked_safe_load: Mock, tmp_path: Path):
+    config_path = tmp_path / 'non_existent.conf'
+    assert not config_path.exists()
 
-    logger.info.assert_called_once_with(
-        "Load logger config: app_mode=<app-mode>, " "config_path=<config-path>, dir=<log-dir>"
-    )
-    print_.assert_called_once_with("Logger config not found in <config-path>. Using default configs.", file=stderr)
-    basic_config.assert_called_once_with(level=logging.INFO, stream=stdout)
+    setup_logging('', Path(''), config_path)
+    assert not mocked_safe_load.called
```

Comment:
The new version of the test does not look clear to me. It is not entirely obvious why `yaml.safe_load` is tested, and is it enough to test just `yaml.safe_load` (and will continue to be enough in the future)

I think it is better to test wether `logging.basicConfig(...)` and/or `ogging.config.dictConfig(...)` were called, something like this:

```python
@patch('tribler.core.logger.logger.logger')
@patch('builtins.print')
@patch('logging.basicConfig')
@patch('logging.config.dictConfig')
def test_setup_logging_no_config(dict_config: Mock, basic_config: Mock, print_: Mock, logger: Mock):
    config_path = MagicMock()
    config_path.exists.return_value = False
    config_path.__str__.return_value = '<config-path>'

    setup_logging('<app-mode>', '<log-dir>', config_path)

    logger.info.assert_called_once_with(
        "Load logger config: app_mode=<app-mode>, " "config_path=<config-path>, dir=<log-dir>"
    )
    print_.assert_called_once_with("Logger config not found in <config-path>. Using default configs.", file=sys.stderr)
    basic_config.assert_called_once_with(level=logging.INFO, stream=sys.stdout)
    dict_config.assert_not_called()
```

Reply:
Both versions of tests (the current and the proposed) do the same. They check what branch of execution is executed in this scenario.

For checking this I use a mock of `yaml.safe_load` function and you offered to check loggers. 

Both versions are unclear, but mine is shorter. 

I could add comments with an explanation of the test.

Reply:
Please take a look at this most recent example of fragility tests based on logging mocks: 
https://github.com/Tribler/tribler/actions/runs/3437151109/jobs/5731538805#step:8:64

Reply:
> Please take a look at this most recent example of fragility tests based on logging mocks

I agree that sometimes logging-related tests may be brittle. Often it means that the test should be rewritten in a more robust way. For example, in the test you are referring to, the correct assertion can be written as:

```python
    for rt in record_tuples:
        assert rt in caplog.record_tuples
```

IMO, the fact that sometimes incorrectly written logging-related checks are brittle does not mean that testing any logging-related functionality should be excluded, especially if the test itself directly relates to logging. `caplog` is an official fixture provided by `pytest` and not a "hack".

Reply:
> Both versions are unclear, but mine is shorter.

The PR version that checks `yaml.safe_load` is intended to check that the code follows the specific code path, but it does not check the final desired state, namely that the `logger.basicConfig(...)` is called.

Tests that check code paths and not the desired state make the codebase more brittle, as it is possible that after some new refactoring, the code path is still executed (to the checked point), but the desired state ceases to be reached.

So, while it is possible to have coverage-only-oriented tests, it is not usually a good idea, in my opinion.

Reply:
I've added comments and checks on the `logging.basicConfig(...)` call.